packages/pigment-css-react/src/processors/sx.ts

```diff
@@ -156,10 +133,136 @@ export class SxProcessor extends BaseProcessor {
-    } else {
-      this.replacer(this.value, false);
+      result = obj;
     }
+
+    /**
+     * Replace the sx call with the transformed result. It works for both JSX and non-JSX calls.
+     *
+     * For example:
+     * <Component sx={_sx({ color: 'red' })} /> to <Component sx={_sx('sd5jss7')} />
+     * <Component sx={_sx({ bgcolor: 'red', color: props.color })} /> to <Component sx={_sx({ className: 'bc1d15y', vars: { 'bc1d15y-0': [props.color, false], }})} />
+     */
+    this.replacer(
+      // @ts-ignore
+      (tagPath: NodePath<CallExpression>) => {
+        return t.callExpression(tagPath.get('callee').node, [result]);
+      },
+      false,
+    );
+
+    /**
+     * For JSX calls, replace the sx prop with runtime sx
+     */
+    this.replacer((_tagPath) => {
+      const tagPath = _tagPath as NodePath<CallExpression>;
+      const jsxElement = tagPath.findParent((p) =>
+        p.isJSXOpeningElement(),
+      ) as NodePath<JSXOpeningElement>;
+      if (!jsxElement) {
+        return tagPath.node;
+      }
+
+      const attributes: any[] = [];
+      let sxAttribute: undefined | NodePath<JSXAttribute>;
+      (jsxElement.get('attributes') as NodePath[]).forEach((attr) => {
+        if (attr.isJSXAttribute() && attr.node.name.name === 'sx') {
+          sxAttribute = attr;
+        } else if (attr.isJSXSpreadAttribute()) {
+          let containRuntimeSx = false;
+          attr.traverse({
+            CallExpression(path) {
+              const callee = path.get('callee');
+              if (callee.isIdentifier() && callee.node.name.startsWith('_sx')) {
+                containRuntimeSx = true;
+              }
+            },
+          });
+          if (!containRuntimeSx) {
+            attributes.push(t.spreadElement(attr.node.argument));
+          }
+        } else if (
+          attr.isJSXAttribute() &&
+          (attr.node.name.name === 'className' || attr.node.name.name === 'style')
+        ) {
+          const value = attr.get('value');
+          if (value.isJSXExpressionContainer()) {
+            attributes.push(
+              t.objectProperty(
+                t.identifier(attr.node.name.name),
+                value.get('expression').node as any,
+              ),
+            );
+          } else {
+            attributes.push(
+              t.objectProperty(t.identifier(attr.node.name.name), attr.node.value as any),
+            );
+          }
+        }
+      });
+      if (sxAttribute) {
+        const expContainer = sxAttribute.get('value');
+        if (expContainer.isJSXExpressionContainer()) {
+          jsxElement.node.attributes.push(
+            t.jsxSpreadAttribute(expContainer.node.expression as Expression),
+          );
+        }
+        sxAttribute.remove();
+      }
+      tagPath.node.arguments.push(t.objectExpression(attributes));
+      return tagPath.node;
+    }, false);
+
+    // For non-JSX calls, replace the sx prop with runtime sx
+    this.replacer((_tagPath) => {
```

Comment:
Why multiple replacer calls? Shouldn't it happen only once?

Reply:
I tried with one replacement but I encountered a weird transformation, it does not work with all the test cases especially conditional and logical expressions, so I split the replacement into 2 steps. First, the existing logic of build-time sx and then replace it with runtime sx spread.

I can combine the replacement for JSX and non-jsx calls.

Reply:
Done.